### PR TITLE
PP-5540 external metadata

### PR DESCRIPTION
**Pull request by jankowiakmaria**
I wasn't quite sure whether to go with moving external_metdata to the transaction details so I prepared two versions:
* current version PR (both commits) shows implementation for keeping external_metadata at the top level
* first commit (only) shows version which moves external_metadata to transaction_details (without changes to db yet)

### CODE REVIEW

**Comment by kbottla** [comment]
I think moving `external_metadata` into `transaction_details` would be ideal and also aligns with the way we wanted to vision `transaction` table

- Anything required for searching / aggregating to be at top level
- Everything else to go `transaction_details`

File: src/test/java/uk/gov/pay/ledger/transaction/dao/TransactionDaoIT.java [hunk 1]
```diff
@@ -88,7 +87,6 @@ public void shouldRetrieveTransactionByExternalIdAndGatewayAccountId() {
         assertThat(transaction.getState(), is(transactionEntity.getState()));
         assertThat(transaction.getEmail(), is(transactionEntity.getEmail()));
         assertThat(transaction.getCardholderName(), is(transactionEntity.getCardholderName()));
-        assertThat(transaction.getExternalMetadata(), is(transactionEntity.getExternalMetadata()));
```
Comment (kbottla) [inline src/test/java/uk/gov/pay/ledger/transaction/dao/TransactionDaoIT.java]: Same here for testing externalMetadata

File: src/test/java/uk/gov/pay/ledger/transaction/dao/TransactionDaoIT.java [hunk 2]
```diff
@@ -45,7 +45,6 @@ public void shouldInsertTransaction() {
         assertThat(retrievedTransaction.getState(), is(transactionEntity.getState()));
         assertThat(retrievedTransaction.getEmail(), is(transactionEntity.getEmail()));
         assertThat(retrievedTransaction.getCardholderName(), is(transactionEntity.getCardholderName()));
-        assertThat(retrievedTransaction.getExternalMetadata(), is(transactionEntity.getExternalMetadata()));
```
Comment (kbottla) [inline src/test/java/uk/gov/pay/ledger/transaction/dao/TransactionDaoIT.java]: We could possibly assert `externalMetadata` from `transactiondetails`

Comment (jankowiakmaria) [reply src/test/java/uk/gov/pay/ledger/transaction/dao/TransactionDaoIT.java]: Didn't expect this to be a blocker for the PR.
Added in below commit.

**Pull request merged**